lib/shopify_api/bulk/query.ex

```diff
@@ -169,4 +211,33 @@ defmodule ShopifyAPI.Bulk.Query do
       _ -> poll(token, bulk_query_id, polling_rate, max_poll, depth + 1)
     end
   end
+
+  defp httpoison_streamed_get(url) do
+    Stream.resource(
+      fn ->
+        HTTPoison.get!(url, %{}, stream_to: self(), async: :once)
+      end,
+      fn %HTTPoison.AsyncResponse{id: id} = resp ->
+        receive do
+          %HTTPoison.AsyncStatus{id: ^id, code: _code} ->
```

Comment:
Any concern over possible non-200 status codes? Will those just result in an empty result stream?